lisa/tools/fio.py

```diff
@@ -61,10 +61,11 @@ def can_install(self) -> bool:
 
     def install(self) -> bool:
         posix_os: Posix = cast(Posix, self.node.os)
-        if posix_os.package_exists("fio"):
-            posix_os.install_packages("fio")
-        else:
+        try:
             self._install_from_src()
+        except Exception as e:
```

Comment:
What's the reason to install from source code firstly? The latest code may not work with old distros.

Reply:
The earlier logic was incorrect.  It is checking if fio is installed with `posix_os.package_exists`. But I found that this instead runs command `['sudo', 'yum', 'list', 'installed', 'fio']` which is just checking if fio is available in the system. If not, it moves to compiling from source which fails on some distro because C11 atomics support is not found. So, in these cases we need to do `yum install fio`. I reorganised the logic so that we rely on compiling first to have more consistency in fio version, but fall back to other version with `yum install` if that fails